app/presenters/check_answers_presenter.rb

```diff
@@ -0,0 +1,62 @@
+class CheckAnswersPresenter
+  include StepsHelper
+
+  def initialize(session_data)
+    @session_data = session_data
+    @model = EstimateModel.new session_data.slice(*EstimateModel::ESTIMATE_ATTRIBUTES.map(&:to_s))
+  end
+
+  def sections
+    data = YAML.load_file(Rails.root.join("app/presenters/check_answers_fields.yml")).with_indifferent_access
+    data[:sections].map { build_section(_1) }.select { _1.subsections.any? }
+  end
+
+  def build_section(section_data)
+    OpenStruct.new(label: section_data[:label],
+                   screen: section_data[:screen],
+                   subsections: build_subsections(section_data))
+  end
+
+  def build_subsections(section_data)
+    return [] if section_data[:subsections].blank?
+
+    section_data[:subsections].map { build_subsection(_1, section_data) }.select { _1.fields.any? }
+  end
+
+  def build_subsection(subsection_data, section_data)
+    OpenStruct.new(label: subsection_data[:label],
+                   screen: subsection_data[:screen],
+                   fields: build_fields(subsection_data[:fields],
+                                        subsection_data[:screen] || section_data[:screen],
+                                        subsection_data[:label] || section_data[:label]))
```

Comment:
We could maybe make this a bit tighter by having two Structs (with keyword_init: true) rather than OpenStruct. OS upsets me as it responds to all methods with nil.